### PR TITLE
fix: pass app context when getting tags for sitemap

### LINKED ISSUE #12: Failed to generate sitemap. Error: TypeError: Cannot destructure property `Tags` of 'undefined' or 'null'.
I'm seeing the following error(s) in MongoDB with job `{ type: "sitemaps/generate" }` using `sitemap-generator 1.2.1`:

Screen Shot 2020-09-04 at 23 55 29

Full error text included in issue title. Appears to be related to the following line of code:

https://github.com/reactioncommerce/api-plugin-sitemap-generator/blob/8f06ee8ec2d96dd0057faf15fd47dc4a2879efb6/src/util/generate-sitemaps.js#L68

To replicate issue run the following mutation on the primary shop and wait for the job to abend:

```graphql
mutation GenerateSitemaps {
  generateSitemaps(
    input: {
      shopId: "cmVhY3Rpb24vc2hvcDphU1lHMnhtZkNGRkZrampCZQ=="
    }
  ), {
    wasJobScheduled
  }
}
```

Then inspect `failures` of most recent `"sitemaps/generate"` job via MongoDB or [Job Control Interface/API](https://github.com/reactioncommerce/reaction-feature-requests/issues/132) if available.

### PR DESCRIPTION
**Pull request by ghost**
fixes #12

Signed-off-by: Josh Habdas <jhabdas@protonmail.com>

Tag sitemaps were failing to be generated due to an error. This pull fixes that error so tag sitemap can generate.

To test do the following:

1. Verify sitemaps are being generated in the `Sitemaps` collection in MongoDB
2. Verify collection contains no record with `handle` eq `sitemap-tags-1.xml`
3. Create or modify tag so its `updatedAt` is later than the `createdAt` of Sitemap `handle` eq `sitemap.xml`
4. Run below mutation to generate Sitemaps
5. Wait for the `sitemaps/generate` job to `complete`
6. Verify Sitemap collection contains record with `handle` eq `sitemap-tags-1.xml`

Mutation to generate Sitemaps:

```graphql
mutation GenerateSitemaps {
  generateSitemaps(
    input: {
      shopId: "cmVhY3Rpb24vc2hvcDphU1lHMhntZkNGRkZrampCZQ=="
    }
  ), {
    wasJobScheduled
  }
}
```

### CODE REVIEW

**Comment by ghost** [comment]
published npm module containing hotfix: https://www.npmjs.com/package/api-plugin-sitemap-generator
production verified: http://www.jamupuri.com/sitemap-tags-1.xml

File: src/util/generate-sitemaps.js [hunk 1]
```diff
@@ -99,7 +99,7 @@ async function generateSitemapsForShop(context, shopId, urlsPerSitemap) {
   const options = { projection: { _id: 1 } };
   const shouldRegenTagSitemaps = hasNoSitemap || !!(await Tags.findOne(selector, options));
   if (shouldRegenTagSitemaps) {
-    const tagSitemapItems = await getTagSitemapItems(shopId);
+    const tagSitemapItems = await getTagSitemapItems(context, shopId);
```
Comment (balibebas) [inline src/util/generate-sitemaps.js:102]: Reference function signature here:

https://github.com/reactioncommerce/api-plugin-sitemap-generator/blob/408a33002ff145ee3e1492d2b263019695e6c688/src/util/generate-sitemaps.js#L216-L224

Comment (ghost) [inline src/util/generate-sitemaps.js:102]: Reference called function signature here:

https://github.com/reactioncommerce/api-plugin-sitemap-generator/blob/408a33002ff145ee3e1492d2b263019695e6c688/src/util/generate-sitemaps.js#L190-L198

**Comment by balibebas** [comment]
+

**Comment by balibebas** [comment]
@focusaurus Could someone review this? It's a very small change with SEO benefit.

**Comment by ghost** [comment]
@mpaktiti Mind reviewing this? Taxonomy sitemaps don't work correctly without it. Thanks!

**Review by mpaktiti — APPROVED** [review]
Looks good! Thanks for fixing that.

**Pull request merged**